src/frontend/classic/components/Viewer.svelte

```diff
@@ -40,11 +44,81 @@
     function toggleFullScreen() {
         if (!document.fullscreenElement) {
             document.documentElement.requestFullscreen();
-        } else {
-            if (document.exitFullscreen) {
-                document.exitFullscreen();
+        } else if (document.exitFullscreen) {
+            document.exitFullscreen();
+        }
+    }
+
+    function loadThumbnail(el: Element) {
+        const div = el as HTMLDivElement;
+        const src = div.getAttribute("data-image");
+
+        if (src) {
+            div.style.backgroundImage = `url('${src}')`;
+        }
+    }
+
+    function loadWide(el: Element) {
+        const img = el as HTMLImageElement;
+        const src = img.getAttribute("data-image");
+
+        if (src) {
+            img.src = src;
+        }
+    }
+
+    function loadImages(currentIndex: number): void {
+        const image = imagesToLoad[currentIndex];
+
+        setTimeout(() => {
+            if (image && image.getAttribute("data-mode") === "thumbnail") {
+                loadThumbnail(image);
+            } else if (image) {
+                loadWide(image);
+            }
+
+            if (currentIndex + 1 < imagesToLoad.length) {
+                loadImages(currentIndex + 1);
+            } else {
+                imagesToLoad = [];
             }
+        }, throttlingDelay);
+    }
+
+    function isImagesToLoadFull(): boolean {
+        return imagesToLoad.length === item.Entries.length;
+    }
+
+    function addElement(el: Element | null) {
+        if (el && el.getAttribute("data-index")) {
+            const index = Number(el.getAttribute("data-index"));
```

Comment:
i see a lot of Dom processing but it could be handled without an index by externalising each image/thumbnail into components that can handle their own context.
I think it would simplify the code